UniqueArray.h

```diff
@@ -23,7 +24,7 @@ class UniqueArray {
     UniqueArray filter(const Filter& f) const;
```

Comment:
Where is the implementation?

Reply:
in the other file, of course :)

Reply:
Oops, I meant the implementation of operator()

Reply:
Oh, I see. The class Filter only exists so that users will be able to derive from it. The function `operator` is virtual - it needs to be overridden by the child class.

You can see an example of usage here: https://github.com/NeatNit/matam_ex3/blob/2be61985860aec9de783d5b273fe5af09075199c/tests/UniqueArrayTest.cpp#L58-L64